### PR TITLE
[YOLOv8] Onnx Export Support

### PR DESCRIPTION
**Pull request by dbogunowicz**
Adds the onnx export support for the YOLOv8 pipeline

### Vanilla Export

```bash
sparseml.ultralytics.export --model /home/ubuntu/damian/sparseml/runs/detect/train80/weights/best.pt
```
```bash
....
2023-02-10 11:43:05 sparseml.pytorch.utils.exporter INFO     Created deployment folder at /home/ubuntu/damian/sparseml/exported/deployment
2023-02-10 11:43:05 sparseml.pytorch.utils.exporter INFO     Saved DetectionModel.onnx in the deployment folder at /home/ubuntu/damian/sparseml/exported/deployment/DetectionModel.onnx
2023-02-10 11:43:05 sparseml.pytorch.utils.exporter INFO     Created config.json file at /home/ubuntu/damian/sparseml/exported/deployment
```
### Sparsified Model Export

```bash
sparseml.ultralytics.export --model /home/ubuntu/damian/sparseml/runs/detect/train57/weights/best.pt
```

```bash
...
2023-02-09 16:36:50 sparseml.pytorch.sparsification.quantization.quantize_qat_export INFO     Converted 63 quantizable Conv ops with weight and bias to ConvInteger and Add
2023-02-09 16:36:51 sparseml.pytorch.utils.exporter INFO     Created deployment folder at /home/ubuntu/damian/sparseml/exported/deployment
2023-02-09 16:36:51 sparseml.pytorch.utils.exporter INFO     Saved DetectionModel.onnx in the deployment folder at /home/ubuntu/damian/sparseml/exported/deployment/DetectionModel.onnx
2023-02-09 16:36:51 sparseml.pytorch.utils.exporter INFO     Created config.json file at /home/ubuntu/damian/sparseml/exported/deployment
Recipe checkpoint detected, saving the recipe to the deployment directory /home/ubuntu/damian/sparseml/exported/deployment
```

### Export + One-Shot

```bash
sparseml.ultralytics.export --model /home/ubuntu/damian/sparseml/runs/detect/train57/weights/best.pt --one-shot one_shot_test.yaml
```

```bash
...
Applying structure from sparseml checkpoint at epoch -1
Loaded previous weights from checkpoint
Source: 'sparseml' detected; Exporting model from SparseML checkpoint...
Detected one-shot recipe: one_shot_test.yaml. Applying it to the model to be exported...
...
2023-02-09 16:39:13 sparseml.pytorch.sparsification.quantization.quantize_qat_export INFO     Converted 63 quantizable Conv ops with weight and bias to ConvInteger and Add
2023-02-09 16:39:14 sparseml.pytorch.utils.exporter INFO     Created deployment folder at /home/ubuntu/damian/sparseml/exported/deployment
2023-02-09 16:39:14 sparseml.pytorch.utils.exporter INFO     Saved DetectionModel.onnx in the deployment folder at /home/ubuntu/damian/sparseml/exported/deployment/DetectionModel.onnx
2023-02-09 16:39:14 sparseml.pytorch.utils.exporter INFO     Created config.json file at /home/ubuntu/damian/sparseml/exported/deployment
Recipe checkpoint detected, saving the recipe to the deployment directory /home/ubuntu/damian/sparseml/exported/deployment
```

### CODE REVIEW

**Review by corey-nm — CHANGES_REQUESTED** [review]
Looks great!

1. Can you test exporting with a yolov8 checkpoint if you haven't already?
2. Can you add a section to the README with instructions? PR description looks great so can probably just copy that

File: src/sparseml/pytorch/utils/exporter.py [hunk 1]
```diff
@@ -248,15 +248,19 @@ def export_torchscript(
...
+        onnx_model_name: Optional[str] = None,
+    ) -> str:
         """
         Create a deployment folder inside the `self._output_dir` directory.
 
         :param labels_to_class_mapping: information about the mapping
             from integer labels to string class names.
             Can be either a string (path to the .json serialized dictionary)
             or a dictionary. Default is None
+        :param onnx_model_name: name of the onnx model file. Default is None
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/exporter.py:262]:
```suggestion
        :param onnx_model_name: name of the onnx model file. Defaults to `model.onnx`
```

**Review by bfineran — APPROVED** [review]
LGTM - echoing corey would also add an example of the expected output directory structure to pr desc/docs

File: src/sparseml/pytorch/yolov8/README.md [hunk 2]
```diff
@@ -55,3 +55,29 @@ for dense training:
...
+
+The resulting exported model will be saved in the appropriate directory structure:
+```bash
+.
+├── deployment
+│    ├── config.json
+│    ├── model.onnx
+│    └── recipe.yaml # optionally, if model contains a sparse checkpoint
+└── model.onnx
+
```
Comment (dbogunowicz) [inline src/sparseml/pytorch/yolov8/README.md:82]:
```suggestion
```

**Review by bfineran — APPROVED** [review]

**Pull request merged**